File: src/de/grobox/liberario/trips/search/TripQuery.java
package de.grobox.liberario.trips.search;

import android.support.annotation.Nullable;

import java.util.Date;

import javax.annotation.ParametersAreNonnullByDefault;

import de.grobox.liberario.favorites.trips.FavoriteTripItem;
import de.grobox.liberario.locations.WrapLocation;

@ParametersAreNonnullByDefault
public class TripQuery {

	public final long uid;
	public final WrapLocation from;
	public final @Nullable WrapLocation via;
	public final WrapLocation to;
	public final Date date;
	public final boolean departure;

	public TripQuery(long uid, WrapLocation from, @Nullable WrapLocation via, WrapLocation to, Date date, boolean departure) {
		this.uid = uid;
		this.from = from;
		this.via = via;
		this.to = to;
		this.date = date;
		this.departure = departure;
	}

	public FavoriteTripItem toFavoriteTripItem() {
		return new FavoriteTripItem(uid, from, via, to);
	}

}

File: app/src/main/java/de/grobox/transportr/trips/TripQuery.kt
package de.grobox.transportr.trips

import de.grobox.transportr.locations.WrapLocation
import de.schildbach.pte.dto.Product
import java.util.*

class TripQuery internal constructor(
        val from: WrapLocation, val via: WrapLocation?, val to: WrapLocation,
        val date: Date,
        departure: Boolean?,
        products: Set<Product>?) {

    val departure = departure != false
    val products: Set<Product> = products ?: EnumSet.allOf(Product::class.java)

}
